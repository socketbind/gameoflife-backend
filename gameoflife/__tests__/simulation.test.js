const simulation = require('../simulation');
const rules = require('../rules');
const habitat = require('../habitat');
const serializer = require('./utils/habitat_snapshot_serializer');

expect.addSnapshotSerializer(serializer);

test('allows creating a basic blinker simulation', () => {
  const blinkerCells = habitat.fromCells([
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
    [0, 1, 1, 1, 0],
    [0, 0, 0, 0, 0],
    [0, 0, 0, 0, 0],
  ]);

  const blinkerSimulation = new simulation.Simulation(blinkerCells, rules.classic, 'Some description');
  expect(blinkerSimulation).toBeDefined();

  const nextState = blinkerSimulation.nextState();
  expect(nextState).toMatchSnapshot('blinkerNextState');
});
