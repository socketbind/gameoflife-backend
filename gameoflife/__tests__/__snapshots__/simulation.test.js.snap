// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows creating a basic blinker simulation: blinkerNextState 1`] = `
_____
__*__
__*__
__*__
_____

`;
